src/Orleans/Core/GrainFactory.cs

```diff
@@ -24,30 +24,37 @@ MIT License
 using System;
 using System.Collections.Concurrent;
 using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Threading.Tasks;
 using Orleans.CodeGeneration;
-using Orleans.Core;
 using Orleans.Runtime;
+using GrainInterfaceData = Orleans.CodeGeneration.GrainInterfaceData;
 
 namespace Orleans
 {
-    using System.Threading.Tasks;
-
     /// <summary>
     /// Factory for accessing grains.
     /// </summary>
     public class GrainFactory : IGrainFactory
     {
         /// <summary>
+        /// The runtime code generator.
+        /// </summary>
+        private static readonly Lazy<IRuntimeCodeGenerator> CodeGenerator = new Lazy<IRuntimeCodeGenerator>(LoadCodeGenerator);
```

Comment:
I think that custom `IGrainFactory` implementation might need to have access to some of the heavy-lifting methods in `GrainFactory`. It's not something which we should do in this commit, though, is it?